src/System.Runtime.Extensions/tests/System/AppDomainTests.cs

```diff
@@ -313,15 +376,30 @@ public void LoadBytes()
         [Fact]
         public void ReflectionOnlyGetAssemblies()
         {
-            Assert.Equal(Array.Empty<Assembly>(), AppDomain.CurrentDomain.ReflectionOnlyGetAssemblies());
+            RemoteInvoke(() => {
```

Comment:
Like stated above in the other comment, tests which would affect others because they are changing states should be invoked inside an RemoteExeuctor instance. It doesn't make sense to "outsource" read operations. We haven't done this anywhere else.